internal/db/store.go

```diff
@@ -131,3 +132,49 @@ func WithTransaction[T any](store Store, fn func(querier ExtendQuerier) (T, erro
 	}
 	return result, store.Commit(tx)
 }
+
+// ProfileRow is an interface row in the profiles table
+type ProfileRow interface {
+	GetProfile() Profile
+	GetEntityProfile() EntityProfile
+}
+
+// GetProfile returns the profile
+func (r ListProfilesByProjectIDAndLabelRow) GetProfile() Profile {
+	return r.Profile
+}
+
+// GetEntityProfile returns the entity profile
+func (r ListProfilesByProjectIDAndLabelRow) GetEntityProfile() EntityProfile {
+	return r.EntityProfile
+}
+
+// GetProfile returns the profile
+func (r ListProfilesByProjectIDRow) GetProfile() Profile {
+	return r.Profile
+}
+
+// GetEntityProfile returns the entity profile
+func (r ListProfilesByProjectIDRow) GetEntityProfile() EntityProfile {
+	return r.EntityProfile
+}
+
+// LabelsFromFilter parses the filter string and populates the IncludeLabels and ExcludeLabels fields
+func (lp *ListProfilesByProjectIDAndLabelParams) LabelsFromFilter(filter string) {
+	// otherwise Split would have returned a slice with one empty string
+	if filter == "" {
+		return
+	}
+
+	for _, label := range strings.Split(filter, ",") {
+		switch {
+		case label == "*":
+			lp.IncludeLabels = append(lp.IncludeLabels, label)
```

Comment:
Should this set `IncludeLabels` to simply the array `["*"]`, and skip further `IncludeLabels`?

Reply:
(We could also do this at after the for loop, with a var that says "was star matched")

I'm wondering if we want this method to return an error for unexpected input.